include/aspect/geometry_model/spherical_chunk.h

```diff
@@ -0,0 +1,205 @@
+         * the first two being the faces perpendicular to the x-axis, the next
+         * two perpendicular to the y-axis, etc.
+         */
+        virtual
+        std::set<types::boundary_id>
+        get_used_boundary_indicators () const;
+
+        /**
+         * Return a mapping from symbolic names of each part of the boundary
+         * to the corresponding boundary indicator. This allows users to
+         * specify *names*, not just *numbers* in their input files when
+         * describing which parts of the boundary have to satisfy which
+         * boundary conditions.
+         *
+         * This geometry returns the map <code>{{"left"->0}, {"right"->1},
+         * {"bottom"->2}, {"top"->3}}</code> in 2d, and <code>{{"left"->0},
+         * {"right"->1}, {"front"->2}, {"back"->3}, {"bottom"->4},
+         * {"top"->5}}</code> in 3d.
+         */
+        virtual
+        std::map<std::string,types::boundary_id>
+        get_symbolic_boundary_names_map () const;
+
+        /**
+         * Return the set of periodic boundaries as described in the input
+         * file.
+         */
+        virtual
+        std::set< std::pair< std::pair<types::boundary_id, types::boundary_id>, unsigned int> >
+        get_periodic_boundary_pairs () const;
+
+
+        /**
+         * Return the typical length scale one would expect of features in
+         * this geometry, assuming realistic parameters.
+         *
+         * We return 1/100th of the diameter of the box.
+         */
+        virtual
+        double length_scale () const;
+
+
+        virtual
+        double depth(const Point<dim> &position) const;
+
+        virtual
+        Point<dim> representative_point(const double depth) const;
+
+        virtual
+        double maximal_depth() const;
+
+        virtual
+        double inner_radius() const;
+
+        virtual
+        double outer_radius() const;
+
+
+        /**
+         * @copydoc Interface::has_curved_elements()
+         *
+         * A box has only straight boundaries and cells, so return false.
+         */
+        virtual
+        bool
+        has_curved_elements() const;
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * Minimum depth, longitude-depth or
+        * longitude-latitude-depth point
+         */
+        Point<dim> point1;
+
+        /**
+               * Maximum depth, longitude-depth or
+         * longitude-latitude-depth point
+               */
+        Point<dim> point2;
+
+        /**
+         * Flag whether the box is periodic in the x-, y-, and z-direction.
+         */
+        bool periodic;
```

Comment:
what about periodicity in some but not all directions? Does periodicity even work?